src/main/java/com/google/devtools/build/lib/analysis/BlazeVersionInfo.java

```diff
@@ -126,7 +131,14 @@ public String getReleaseName() {
    */
   public String getVersion() {
     String buildLabel = buildData.get(BUILD_LABEL);
-    return buildLabel != null ? buildLabel : "";
+    if (buildLabel != null) {
+      return buildLabel;
+    }
+    String override = System.getenv(BAZEL_DEV_VERSION_OVERRIDE_ENV_VAR);
```

Comment:
I wonder if there is a nice way to notice users that this env var is in effect?

Reply:
That's unfortunately not easy as a number of call sites don't have an `EventHandler` handy. It only takes effect on dev versions, which I hope would mitigate most realistic cases in which one would miss this being set accidentally.

Reply:
Makes sense!